TypeScript v2.4 caused validation errors on Generics

[Announcing TypeScript 2.4](https://blogs.msdn.microsoft.com/typescript/2017/06/27/announcing-typescript-2-4/)

TypeScript 2.4 introduces "Stricter checking for generic functions", which I think is what's causing these errors.

e.g. the interface defines ScreenSnippet.capture() as returning a `Promise<string>`, but the actual function returns just a `Promise`

Example error:
```
src/screen-snippet.ts(1,7): error TS2420: Class 'ScreenSnippet' incorrectly implements interface 'ssf.ScreenSnippet'.
  Types of property 'capture' are incompatible.
    Type '() => Promise<{}>' is not assignable to type '() => Promise<string>'.
      Type 'Promise<{}>' is not assignable to type 'Promise<string>'.
        Type '{}' is not assignable to type 'string'.
```

--- a/packages/api-browser/src/window.ts
+++ b/packages/api-browser/src/window.ts
@@ -80,7 +80,7 @@ class Window implements ssf.WindowCore {
   }
 
   getParentWindow() {
-    return new Promise(resolve => {
+    return new Promise<Window>(resolve => {
       let newWin = null;
       if (window.opener) {
         newWin = new Window(null);

--- a/packages/api-electron/src/preload/screen-snippet.ts
+++ b/packages/api-electron/src/preload/screen-snippet.ts
@@ -2,7 +2,7 @@ const remote = require('electron').remote;
 
 class ScreenSnippet implements ssf.ScreenSnippet {
   capture() {
-    return new Promise((resolve) => {
+    return new Promise<string>((resolve) => {
       remote.getCurrentWindow().capturePage((image: any) => {
         const dataUri = 'data:image/png;base64,' + image.toPng().toString('base64');
         resolve(dataUri);

--- a/packages/api-openfin/package.json
+++ b/packages/api-openfin/package.json
@@ -16,9 +16,9 @@
   "author": "",
   "license": "Apache-2.0",
   "devDependencies": {
+    "containerjs-api-specification": "0.0.2",
     "copyfiles": "^1.2.0",
     "rimraf": "^2.6.1",
-    "containerjs-api-specification": "0.0.2",
     "rollup": "^0.41.6",
     "typescript": "^2.3.2"
   },

--- a/packages/api-openfin/src/screen-snippet.ts
+++ b/packages/api-openfin/src/screen-snippet.ts
@@ -1,6 +1,6 @@
 class ScreenSnippet implements ssf.ScreenSnippet {
   capture() {
-    return new Promise((resolve, reject) => {
+    return new Promise<string>((resolve, reject) => {
       fin.desktop.Window.getCurrent()
         .getSnapshot((snapshot) => { resolve('data:image/png;base64,' + snapshot); }, reject);
     });

--- a/packages/api-openfin/src/window.ts
+++ b/packages/api-openfin/src/window.ts
@@ -265,7 +265,7 @@ class Window implements ssf.Window {
   }
 
   getParentWindow() {
-    return new Promise((resolve, reject) => {
+    return new Promise<Window>((resolve, reject) => {
       if (this.innerWindow) {
         let parent = null;
         if (window.opener) {

--- a/packages/api-specification/interface.ts
+++ b/packages/api-specification/interface.ts
@@ -203,7 +203,7 @@ declare namespace ssf {
      * Get the parent of the window. Null will be returned if the window has no parent.
      * @returns The parent window.
      */
-    getParentWindow(): Promise<Window>;
+    getParentWindow(): Promise<Window|WindowCore>;
 
     /**
      * Get the position of the window.

